### ISSUE TITLE
[Train][Feature][Train/RLlib Integration] Add flags to disable creating log directories

### ISSUE THREAD

**Issue by avnishn** [ray-project/ray#22261]
### Search before asking

- [X] I had searched in the [issues](https://github.com/ray-project/ray/issues) and found no similar feature requirement.

### Description

In order to prototype tune/rllib integration and assess its performance, we would need to disable expensive file io operations that aren't necessary during the training loop. 

Right now, I can probably redirect training logs to /tmp directory, but I don't know how expensive these file operations are. Perhaps they aren't all that expensive if I don't actually add any log data via `train.report`. @matthewdeng @amogkam wdyt?

### Use case

RLlib/tune integration

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

**Comment by matthewdeng** [comment]
`train.report()` calls aren't actually logged unless you pass in a logging Callback (e.g. `JsonLoggerCallback`).

**Comment by avnishn** [comment]
Should I close this issue then? Are logging directories/any file io ops going to happen in the case that I don't use any logging callbacks?

**Comment by amogkam** [comment]
That’s right train will create the directories, but nothing actually gets added to them unless you specify a callback or checkpoint.

But let’s keep this open because we should still provide a way to disable even the initial creation of the directories.

**Comment by amogkam** [comment]
To me, 1 and raising a meaningful error seems like the best option.

As a user, if I explicitly disable dir creation, there's not reason I should expect logging and checkpointing to work.

**Comment by amogkam** [comment]
cc @JiahaoYao

**Comment by cszhu** [comment]
Hello! This P3 issue has seen no activity in at least a year. It will be closed in 2 weeks as part of ongoing cleanup efforts.

Please remove the `pending-cleanup` label if you believe this issue should remain open.

Thanks for contributing to Ray!
